apps/users/models.py

```diff
@@ -361,7 +347,12 @@ def source(self):
 
     @property
     def name(self):
-        return smart_unicode(self.display_name or self.username)
+        if self.display_name:
+            return smart_unicode(self.display_name)
+        elif self.username.startswith('anonymous-'):
```

Comment:
I agree. I'd rather have a boolean column or allow it to be `None` but this felt better than adding a migration to the table. I could update it to match a regex of `anonymous-[a-f0-9]{32}`. Which would be pretty unlikely.